torch/csrc/jit/codegen/cuda/ops/composite.cpp

```diff
@@ -29,6 +29,13 @@ ForwardDropoutResult dropout(TensorView* x, Val* prob, Val* scale) {
 
   auto rand_vals = randlike(x);
```

Comment:
Alternatively, we could replace ones with zeros in `rand_vals` to change the interval from (0.0, 1.0] to [0.0, 1.0).
```suggestion
  auto rand_vals = randlike(x);
  auto rand_vals_new_interval = where(eq(rand_vals, IrBuilder::create<Double>(1.0)), IrBuilder::create<Double>(0.0), rand_vals);
```
Not sure, if using where or bitwise functions is better.

Reply:
`where` looks cleaner, I also think it should handle broadcast properly as well~ Let me try that~

Reply:
Using `where` inside `randlike` would also fix https://github.com/csarofeen/pytorch/issues/1807.

Reply:
That does sound right as well, are we pushing towards that fix?

Having no idea how RNG works. To my naive mind, it only messes with some corner distribution when we squash 1->0, so I'm somewhat comfortable that nobody would notice the difference of the distribution....

Reply:
I would push towards that fix.

It's difficult to notice this kind of difference, like the problem of dropout was there for some time already.
It's fine swapping 1.0 with 0.0 because mathematically every number has an equal probability to be generated and we won't mess up any mathematical properties.

CuPy does the same:
https://github.com/cupy/cupy/blob/18fab32f3d347fc13f86821cc72964515f6e5c4f/cupy/random/_generator.py#L619
```
    _mod1_kernel = _core.ElementwiseKernel(
        '', 'T x', 'x = (x == (T)1) ? 0 : x', 'cupy_random_x_mod_1')
```

Reply:
Updated to that. I'm fine with the change. A trivial question doesn't clapping 1 to 0 means that we'll have twice as many 0s as other numbers? But that's just a minor thing so not really trying to dive into that rabbit hole.

Does the update look better?

Reply:
There wouldn't be twice as many 0s, because there should be none generated by the Philox RNG (implemented here https://github.com/csarofeen/pytorch/blob/282c42902bff07f759cddbbe619249cf5e7c5281/torch/csrc/jit/codegen/cuda/runtime/random_numbers.cu#L16), at least usually it's the case that 0 is excluded.